### DISCUSSION TITLE
Is it possible to have a module which will accept input?

### DISCUSSION THREAD

**Discussion by WillAdams** [DSchroer/dslcad#17] [General]
As an example, one of the modules which I use in the gcodepreview library for OpenSCAD which I've been working on is:

    module rapid(bx, by, bz, ex, ey, ez) {
    if (generategcode == true) {
            writeln("(rapid)");
    	    writeln("G0 X",ex," Y", ey, "Z", ez);
    }
    hull(){
        translate([bx, by, bz]){
          select_tool(102);
        }
        translate([ex, ey, ez]){
          select_tool(102);
        }
      }
    }

and I'm not seeing how to achieve something similar in DSLcad --- I think this is the last piece I'd need to start in earnest (my apologies for piece-mealing this).

**Comment by DSchroer** [comment]
Hey. Yes there is a way to do this. Right now each file is a module and each variable can be considered an input. So for your example I would do:
```
// rapid.ds
var b;
var e;
"(rapid)\n"
+ formatln(message="GO X{x} Y{y} Z{z}", x=e.x, y=e.y, z=e.z);
```

Then in another file you can use it like this:
```
./rapid(b=point(x=1, y=1, z=1));
```
